retrofit2-clients/src/main/java/com/palantir/remoting2/retrofit2/MultiServerRetryInterceptor.java

```diff
@@ -86,12 +86,12 @@ public Response intercept(Chain chain) throws IOException {
     }
 
     private Request redirectRequest(Request request, String redirectToUri) {
-        String requestUri = request.url().toString();
+        HttpUrl requestUrl = request.url();
 
         String matchingUri = null;
         // Find which server from `uris` is used in the current request, then ...
         for (String uri : uris) {
-            if (requestUri.startsWith(uri)) {
+            if (doServersMatch(requestUrl, HttpUrl.parse(uri))) {
```

Comment:
ah, swallowed some characters, sorry; wondering if the class should keep a `List<HttpUrl>` instead of a `List<String>`?